src/EditorFeatures/Core/Implementation/CodeFixes/CodeFixService.cs

```diff
@@ -174,15 +174,21 @@ public async Task<ImmutableArray<CodeFixCollection>> GetFixesAsync(Document docu
 
                 cancellationToken.ThrowIfCancellationRequested();
 
-                aggregatedDiagnostics ??= new SortedDictionary<TextSpan, SortedSet<DiagnosticData>>();
-                aggregatedDiagnostics.GetOrAdd(diagnostic.GetTextSpan(), _ => new SortedSet<DiagnosticData>(s_diagnosticDataComparerById)).Add(diagnostic);
+                aggregatedDiagnostics ??= new SortedDictionary<TextSpan, List<DiagnosticData>>();
+                aggregatedDiagnostics.GetOrAdd(diagnostic.GetTextSpan(), _ => new List<DiagnosticData>()).Add(diagnostic);
             }
 
             if (aggregatedDiagnostics == null)
             {
                 return ImmutableArray<CodeFixCollection>.Empty;
             }
 
+            // Order diagnostics by DiagnosticId so the fixes are in a deterministic order.
+            foreach (var diagnosticsWithSpan in aggregatedDiagnostics.Values)
+            {
+                diagnosticsWithSpan.Sort(s_diagnosticDataComparisonById);
```

Comment:
`List<T>.Sort` is an unstable sort, so for cases of duplicate diagnostics at the same location, they could move around with each pass. It would be good to use LINQ's `OrderBy` for this operation, which is a stable sort.

Reply:
Hmm I am not sure what would be a desirable diagnostic/fix order for such duplicate diagnostics though. Do we even have code fixes that generate different code actions for such duplicate diagnostics? I doubt if we do. I am going to wait for a realistic use case for which a unit test can be written before making a change.